editor.planx.uk/src/@planx/components/shared/Schema/Fields.tsx

```diff
@@ -17,45 +26,49 @@ import ErrorWrapper from "ui/shared/ErrorWrapper";
 import Input from "ui/shared/Input";
 import InputRowLabel from "ui/shared/InputRowLabel";
 
-import { DESCRIPTION_TEXT, ERROR_MESSAGE } from "../../shared/constants";
-import BasicRadio from "../../shared/Radio/BasicRadio";
-import type {
-  ChecklistField,
-  DateField,
-  NumberField,
-  QuestionField,
-  TextField,
-} from "../model";
-import { useListContext } from "./Context";
+import { DESCRIPTION_TEXT, ERROR_MESSAGE } from "../constants";
+import BasicRadio from "../Radio/BasicRadio";
+
+type Props<T extends Field> = {
+  formik: FormikProps<SchemaData>
+  activeIndex: number;
+} & T;
 
-type Props<T> = T & { id: string };
+/**
+ * Helper function to get shared props derived from `Field` and `props.formik`
+ */
+const getFieldProps = <T extends Field>(props: Props<T>) => ({
```

Comment:
This function allows the components below to have a cleaner prop syntax and ensures we handle all of these in a standardised way.

Reply:
Helpful!